src/models/generic/WorkflowModel.ts

```diff
@@ -1029,4 +1052,176 @@ export abstract class WorkflowModel extends ValidationBase implements Serializab
+
+        // If type is the same (string -> string...) or type is array and items has the same type ([string] -> [string]...)
+        if (destinationType.type === sourceType.type && destinationType.items === sourceType.items) {
+
+            if (destinationType.type === "File" || destinationType.items === "File") {
+                isValid = checkFileTypes();
+            }
+
+        } else {
+
+            // If destination is array of items of type source (string -> [string]...)
+            if (destinationType.items === sourceType.type) {
+                isValid = checkFileTypes();
+            } else {
+                isValid = false;
+            }
+        }
+
+        // Whether to update connections or not (false when called from connectSource function)
+        if (updateConnection) {
+            this.connections.filter((c) =>
+                    c.isVisible && (c.destination.id === destination.connectionId && c.source.id === source.connectionId))
+                .forEach((c) => {
+                c.isValid = isValid;
+            });
+        }
+
+        if (!isValid) {
+            // Set warning
+            destination.updateValidity({[destination.loc + ".sources[" + source.sourceId + "]"]: {
+                message: `invalid connection (type or file types mismatch)`,
+                type: "warning"
+            }});
+        }
+
+        return isValid;
+    }
+
+    /**
+     * Validate all connections made with given destination
+     */
+    private validateDestination(destination: WorkflowOutputParameterModel | WorkflowStepInputModel, updateConnection = false) {
+
+        destination.cleanValidity();
+
+        // Find all sources connected to given destination
+        const sources = this.connections.filter((connection) => {
+            return connection.isVisible && (connection.destination.id === destination.connectionId);
+        }).map((connection) => {
+            return this.findById(connection.source.id);
+        });
+
+        // Validate all connections
+        sources.forEach((source) => {
+           this.validateConnection(destination, source, updateConnection);
+        });
+
+    }
+
+    /**
+     * Validate all connections made with given IO port
+     */
+    validateConnectionsForIOPort(port: WorkflowOutputParameterModel | WorkflowInputParameterModel) {
+
+        const isInput = port instanceof WorkflowInputParameterModel;
+
+        if (!isInput) {
+            // If port is output
+            this.validateDestination(port as WorkflowOutputParameterModel, true);
+
+        } else {
+            const destinations = this.connections.filter((connection) => {
+                return connection.isVisible && (connection.source.id === port.connectionId);
+            }).map((connection) => {
+                return this.findById(connection.destination.id);
+            });
+
+            destinations.forEach(destination => {
+                this.validateDestination(destination, true);
```

Comment:
Calling `validateDestination` like this will validate connections that hadn't changed, in case those destinations are connected to sources other than the input being changed. 

Is there a reason not to call `validateConnection` here, considering we know exactly which source and destinations need to be updated?